# Project 1

In this first project you will create a framework to scope out data science projects. This framework will provide you with a guide to develop a well-articulated problem statement and analysis plan that will be robust and reproducible.

### Read and evaluate the following problem statement: 
Determine which students will get admitted to the University of Eddie, using their academic scoring (GRE and GPA) and the prestige of their high school.

#### 1. What is the outcome?

Answer: Admission (boolean)

#### 2. What are the predictors/covariates? 

Answer: gre, gpa, prestige

#### 3. How many examples are in the data set?

Answer: 400

#### 4. What is the hypothesis?

Answer: Higher GRE, GPA, and prestige will lead to higher probability of admissions

## Let's get started with our dataset

#### 1. Create a data dictionary 

Answer: 

Variable | Description | Type of Variable
---| ---| ---
gre | thing in unit X | continuous 
gpa | thing in unit X | continuous 
prestige | 1 = low 2 = medium 3 = high | categorical


We would like to explore the association between admit and each of the variables 

In [41]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib as plt
%matplotlib inline

In [42]:
df = pd.read_csv('/Users/edwardhuang/Desktop/DS-SF-42/unit-project/project-1/assets/admissions.csv')

In [43]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 400 entries, 0 to 399
Data columns (total 4 columns):
admit       400 non-null int64
gre         398 non-null float64
gpa         398 non-null float64
prestige    399 non-null float64
dtypes: float64(3), int64(1)
memory usage: 12.6 KB


There are 400 examples of admissions, with 4 columns (admit, gre, gpa, prestige)

In [44]:
df.isnull().sum()

admit       0
gre         2
gpa         2
prestige    1
dtype: int64

In [45]:
gre = df[df['gre'].isnull()]
gpa = df[df['gpa'].isnull()]
prestige = df[df['prestige'].isnull()]

In [46]:
pd.concat([gre,gpa,prestige]).index.unique()

Int64Index([187, 212, 236], dtype='int64')

187, 212, 236 are all rows that have null values somewhere. They will be dropped.

In [47]:
clean = df.dropna()

In [48]:
clean.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 397 entries, 0 to 399
Data columns (total 4 columns):
admit       397 non-null int64
gre         397 non-null float64
gpa         397 non-null float64
prestige    397 non-null float64
dtypes: float64(3), int64(1)
memory usage: 15.5 KB


## Let's find out which X variable is most correlated to the admit outcome!

In [50]:
df.corr()

admit       gre       gpa  prestige
admit     1.000000  0.182919  0.175952 -0.241355
gre       0.182919  1.000000  0.382408 -0.124533
gpa       0.175952  0.382408  1.000000 -0.059031
prestige -0.241355 -0.124533 -0.059031  1.000000

#### 2. What is the outcome?

Answer:

#### 3. What are the predictors/covariates? 

Answer:

#### 4. What timeframe is this data relevent for?

Answer:

#### 4. What is the hypothesis?

Answer:

    Using the above information, write a well-formed problem statement. 


## Problem Statement

### Exploratory Analysis Plan

Using the lab from a class as a guide, create an exploratory analysis plan. 

#### 1. What are the goals of the exploratory analysis? 

Answer: 

#### 2a. What are the assumptions of the distribution of data? 

Answer:

#### 2b. How will determine the distribution of your data? 

Answer: 

#### 3a. How might outliers impact your analysis? 

Answer: 

#### 3b. How will you test for outliers? 

Answer:

#### 4a. What is colinearity? 

Answer: 

#### 4b. How will you test for colinearity? 

Answer:

#### 5. What is your exploratory analysis plan?
Using the above information, write an exploratory analysis plan that would allow you or a colleague to reproduce your analysis 1 year from now. 

Answer: 

## Bonus Questions:
1. Outline your analysis method for predicting your outcome
2. Write an alternative problem statement for your dataset
3. Articulate the assumptions and risks of the alternative model